picard/config.py

```diff
@@ -341,12 +319,51 @@ class ListOption(Option):
 setting = None
 persist = None
 
+PURGE_INTERVAL = 60000
+_thread_configs = {}
+_thread_config_lock = threading.RLock()
+
 
-def _setup(app, filename=None):
+def setup_config(app, filename=None):
     global config, setting, persist
     if filename is None:
         config = Config.from_app(app)
     else:
         config = Config.from_file(app, filename)
+    _thread_configs[threading.get_ident()] = config
     setting = config.setting
     persist = config.persist
+    QtCore.QTimer.singleShot(PURGE_INTERVAL, purge_config_instances)
+
+
+def get_config():
+    """Returns a config object for the current thread.
+
+    Config objects for threads are
+    """
+    thread_id = threading.get_ident()
+    thread_config = _thread_configs.get(thread_id)
+    if not thread_config:
+        _thread_config_lock.acquire()
+        try:
```

Comment:
Is there a specific reason to use try/finally here? It may hide exceptions.

Is it possible to just use:

```python
with _thread_config_lock:
    do_stuff
```

and if exceptions are expected:

```python
with _thread_config_lock:
    try:
        do_stuff
    except Exception as e:
        log_this
```

Reply:
The locks from threading do not implement a context manager, so you cannot use `with`. But

```python
_thread_config_lock.acquire()
try:
    do_stuff
finally:
    _thread_config_lock.release()
```

is functional equivalent to

```python
with _thread_config_lock:
    do_stuff
```

It doesn't hide exceptions, as it doesn't catch any. But it ensures the lock is released. It shouldn't matter here. No exceptions are expected here, and should one arise the main thread would exit anyway, so no real need to release the lock. But I think it is good practice to follow this pattern when dealing with locks.

Reply:
Damn, I shouldn't review stuff without enough coffee in my body...